### ISSUE TITLE
Flexible content field group

### ISSUE THREAD

**Issue by cmwwebfx** [michaeluno/admin-page-framework#180]
Hi Michael,

Happy new year..

I have used a plugin before for creating custom fields for a small project, and it was called ACF (Advanced Custom Fields) by Elliot Condon.

While I used this plugin for creating a simple layout, I explored the options they had available in the PRO version. What I really loved was something called "Flexible Content Field". 

http://www.advancedcustomfields.com/resources/flexible-content/

This was really awesome since it allowed me to make a set of grouped fields, then add them to this area that I wanted. I used this to make a pretty complex , yet easy to use selection of groups for the end user.

Is this something you might consider to add to Admin Page Framework? Here are some screenshots of how I have used it, and you can see how this is quite beneficial to have such an option.

![add a flexible content field](https://cloud.githubusercontent.com/assets/1312880/5591413/18e8920e-91aa-11e4-9744-8630fcc2eb81.jpg)

![flexible content field](https://cloud.githubusercontent.com/assets/1312880/5591415/36f2ef60-91aa-11e4-92b3-a4d749d4afe5.jpg)

![flexible content field dragable](https://cloud.githubusercontent.com/assets/1312880/5591416/3a79ee2c-91aa-11e4-9656-f9ae00f86776.jpg)

**Comment by michaeluno** [comment]
Indeed, the ability to layout mixed field types in a single field is something I've been thinking how to achieve. Before this, the ability of nested fields should be carried out. Can you suggest the syntax for it, like how the user sets the arguments and what kind of method they would use?

The form builder functionality should be considered separately. And it may be implemented with a admin page builder. If you need those, post new topics.

**Comment by cmwwebfx** [comment]
Form builder is of no real concern right now, but that would be cool if there was such a thing. Again, not needed since this is a development type framework. I am assuming you were referring to a builder like ACF?

As for the idea and how the dev will set these arguments.

Create repeatable/sortable groups.
Develop new field ("select_group") to place in an array of which repeatable groups to include.

**Comment by cmwwebfx** [comment]
Hi Michael, 
Thank you for responding to my other query. I would happily [buy into the dev licence](http://michaeluno.jp/checkout/?edd_action=add_to_cart&download_id=3216&edd_options[price_id]=2) if this feature #180, #152 and #170 if they were included as paid features.

**Comment by michaeluno** [comment]
This is now included in the [3.8.0/main](https://github.com/michaeluno/admin-page-framework/tree/3.8.0/main) branch. 

![inline_mixed_fields](https://cloud.githubusercontent.com/assets/3408158/15427991/cab9a966-1ed3-11e6-9b27-d08ce5a123cd.png)

### Usage

Use the `content` argument and set a string with placeholders to the first element. The format of placeholders is the same as the one used for the [sprintf()](http://php.net/manual/en/function.sprintf.php) function. 

In the below example, `%1$s` is replaced with the first following field and `%2$s` is replaced with the second field and so on. And in order to set a width, at the moment, you need to set it with the `attributes` argument with the `fieldset` key.

```php
            array(
                'field_id'      => 'inline_mixed',
                'title'         => __( 'Inline Mixed', 'admin-page-framework-loader' ),            
                'content'       => array(
                    '%1$s %2$s %3$s %4$s',
                    array(
                        'field_id'        => 'enable',
                        'type'            => 'checkbox',
                        'label'           => __( 'Do something in', 'admin-page-framework-loader' ),
                    ),                
                    array(
                        'field_id'        => 'interval',
                        'type'            => 'number',
                        'label_min_width' => '',
                        'default'         => 3,
                        'attributes'      => array(
                            'style' => 'width: 80px;',
                        ),
                    ),    
                    array(
                        'field_id'        => 'interval_unit',
                        'type'            => 'select',
                        'label_min_width' => '',
                        'label'           => array(
                            'hour'    => __( 'hours', 'admin-page-framework-loader' ),
                            'day'     => __( 'days', 'admin-page-framework-loader' ),
                            'week'    => __( 'weeks', 'admin-page-framework-loader' ),
                        ),                    
                        'default'         => 'day',
                    ),
                    array(
                        'field_id'      => '_text',
                        'content'       => __( 'to do something else.', 'admin-page-framework-loader' ),
                    ),                                 
                ),
            ),
            array(
                'field_id'      => 'shipping_address',
                'title'         => __( 'Shipping Information', 'admin-page-framework-loader' ),
                'repeatable'    => true,
                'sortable'      => true,
                'content'       => array(
                    '%1$s%2$s%3$s%4$s%5$s%6$s%7$s%8$s%9$s',
                    array(
                        'field_id'      => 'first_name',
                        'type'          => 'text',
                        'title'         => __( 'First Name', 'admin-page-framework-loader' ),
                        'attributes'    => array(
                            'fieldset'  => array(
                                'style'  => 'width: 50%',
                            ),
                        ),                        
                    ),
                    array(
                        'field_id'      => 'last_name',
                        'type'          => 'text',
                        'title'         => __( 'Last Name', 'admin-page-framework-loader' ),
                        'attributes'    => array(
                            'fieldset'  => array(
                                'style'  => 'width: 50%',
                            ),
                      
                        ),
                    ),
                    array(
                        'field_id'      => 'mailing_address',
                        'type'          => 'text',
                        'title'         => __( 'Street Address', 'admin-page-framework-loader' ),
                        'attributes'    => array(
                            'fieldset'  => array(
                                'style'  => 'width: 100%'
                            ),
                        ),                                        
                    ),
                    array(
                        'field_id'      => 'city',
                        'type'          => 'text',
                        'title'         => __( 'City/Town', 'admin-page-framework-loader' ),
                        'attributes'    => array(
                            'fieldset'  => array(
                                'style'  => 'width: 38%'
                            ),
                        ),
                    ),
                    array(
                        'field_id'      => 'state',
                        'type'          => 'text',
                        'title'         => __( 'State/Province', 'admin-page-framework-loader' ),
                        'attributes'    => array(
                            'fieldset'  => array(
                                'style'  => 'width: 38%'
                            ),
                        ),                
                    ),
                    array(
                        'field_id'      => 'zip',
                        'type'          => 'text',
                        'title'         => __( 'Zip/Postal Code', 'admin-page-framework-loader' ),
                        'attributes'    => array(
                            'fieldset'  => array(
                                'style'  => 'width: 24%; clear:right;'
                            ),
                        ),  
                    ),                    
                    array(
                        'field_id'      => 'telephone',
                        'type'          => 'text',
                        'title'         => __( 'Tel. No.', 'admin-page-framework-loader' ),
                        'attributes'    => array(
                            'fieldset'  => array(
                                'style'  => 'width: 33%'
                            ),
                        ),
                    ),
                    array(
                        'field_id'      => 'fax',
                        'type'          => 'text',
                        'title'         => __( 'Fax No.', 'admin-page-framework-loader' ),
                        'attributes'    => array(
                            'fieldset'  => array(
                                'style'  => 'width: 33%'
                            ),
                        ),            
                    ),
                    array(
                        'field_id'      => 'email',
                        'type'          => 'text',
                        'title'         => __( 'Email', 'admin-page-framework-loader' ),
                        'attributes'    => array(
                            'fieldset'  => array(
                                'style'  => 'width: 34%'
                            ),
                        ),                  
                    ),                         
                ),                                
            )
```

Example code can be found [here](https://github.com/michaeluno/admin-page-framework/blob/3.8.0/main/example/admin_page/advanced_usage/mixed/APF_Demo_AdvancedUsage_Mixed_Inline.php). Please try it and report issues. Thanks.

**Comment by michaeluno** [comment]
Now I think using placeholders is not useful as the embedded field outputs are all enclosed in structured container elements and normal text set along with the placeholders results in having different CSS rules than those embedded field outputs and this requires the user to adjust the style. 

So I'm going to change the above syntax to use  a new field type,`inline_mixed` by deprecating the string element with placeholders set in the `content` argument.
